### ISSUE TITLE
Setting NODE_ENV to production triples bundle size

### ISSUE THREAD

**Issue by ryands17** [serverless/serverless#9134]
As per this discussion: https://github.com/serverless/serverless/discussions/9123 I have a [repo](https://github.com/ryands17/serverless-prisma/) deploying a Serverless GraphQL app. 

Using `sls package` with `NODE_ENV` set to `production` gives almost 3 times the bundle size (screenshots in the above discussion).

<b>Installed version</b>

```
Framework Core: 2.30.0 (local)
Plugin: 4.5.0
SDK: 4.2.0
Components: 3.7.3
```

**Comment by pgrzesik** [comment]
Hello @ryands17 :wave: Thanks a lot for reporting your issue and sorry you've run into trouble. Internally we're using `npm ls` command to determine the list of packages for exclusion, however, I've noticed a weird behavior of `npm ls` when `dev/prod` flags are used together with `NODE_ENV` - as a workaround, is `NODE_ENV=production` required in your case? Does it also happen if you don't set it?

**Comment by ryands17** [comment]
`NODE_ENV` production is required in my use case for the Lambda, so currently I am using `dotenv` and an `.env` file as a workaround :)

It bundles it totally fine if `NODE_ENV` is not set. Only setting it to `production` causes an issue.

**Comment by pgrzesik** [comment]
Fix will be released with the next release which will most likely be out early next week - I've confirmed locally on your example that it should work without issues. Thanks for bringing this to our attention @ryands17 :bow:

**Comment by pgrzesik** [comment]
Just FYI - we've released `2.31.0` version that should now correctly exclude dependencies when `NODE_ENV` is set :+1:
